test/core-language/test.jam

```diff
@@ -1505,11 +1505,32 @@ check-equal shell : "value\n"   : [ SHELL $(c) ] ;
 check-equal shell : ""          : [ SHELL $(c) : no-output ] ;
 check-equal shell : "value\n" 0 : [ SHELL $(c) : exit-status ] ;
 check-equal shell : ""      0   : [ SHELL $(c) : no-output : exit-status ] ;
+check-equal shell : ""      0   : [ SHELL $(c) : no-output : exit-status : strip-eol ] ;
 check-equal command : "value\n"   : [ COMMAND $(c) ] ;
 check-equal command : ""          : [ COMMAND $(c) : no-output ] ;
 check-equal command : "value\n" 0 : [ COMMAND $(c) : exit-status ] ;
 check-equal command : ""      0   : [ COMMAND $(c) : no-output : exit-status ] ;
 
+# buffered output
+
+local expected = "When the shell output buffer splits on whitespace, the whitespace shouldn't be trimmed.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             end." ;
```

Comment:
This string is the key component to these tests. It's hard to see, but there is a large amount of whitespace between `trimmed.` and `end.`. The length of this string is 1096 bytes. Provided that the output is split in 1024 byte regions, the first chunk will have a large amount of whitespace trailing after `trimmed.`.

The more common error case (as I've seen), is when trying to parse something like the output of a `pkg-config` command and the whitespace boundary just _happens_ to be between a set of cflags or linker flags.